### PR TITLE
Slab ground heat transfer program has an exception during execution

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
 - Followup to #8333
 - Suggested change similar to #9356

image

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by rraustad** [comment]
Some CMake expert will have to tell me where the mistakes are.

**Comment by rraustad** [comment]
Thanks for the suggestion @mjwitte that seemed to have worked.

image

**Comment by rraustad** [comment]
Comparing all 5ZoneAirCooledWithSlab results files using V22-1-0 and this branch's release version shows no diffs (i.e., csv, eio, bnd, err, etc.). So not sure what this corrected.

**Comment by rraustad** [comment]
@jmarrec would you please compare this change to #9356 since I am not familiar with CMake. Also, the CMakeLists.txt file in ...\src\Slab and ...\src\Basement are not as similar as I would have thought so I am not sure if something is missing in this branch.

File: src/Slab/CMakeLists.txt [hunk 1]
```diff
@@ -33,7 +38,7 @@ if(NOT UNIX)  # Need to reinclude it on Windows
 endif()
 get_target_property(F_INTERFACE_COMPILE_FLAGS fortran_project_options INTERFACE_COMPILE_OPTIONS)
 message(DEBUG "Slab: F_INTERFACE_COMPILE_FLAGS=${F_INTERFACE_COMPILE_FLAGS}")
-target_link_libraries(Slab PRIVATE fortran_project_options)
+target_link_libraries(Slab PRIVATE SlabOptions fortran_project_options)
 
 install(TARGETS Slab DESTINATION PreProcess/GrndTempCalc)
 if(NOT ${STATIC_EXE})
```
Comment (rraustad) [inline src/Slab/CMakeLists.txt]: @jmarrec CMakeLists.txt for ...\src\Basement shows this as if(NOT ${FORTRAN_STATIC_EXE}). Should that key word also be used here? Searching for FORTRAN_STATIC_EXE in this repo shows all other fortran projects use that word. Slab is the only one that uses STATIC_EXE.

Comment (jmarrec) [reply src/Slab/CMakeLists.txt]:
```suggestion
if(NOT ${FORTRAN_STATIC_EXE})
```

**Review by jmarrec — APPROVED** [review]

File: src/Slab/CMakeLists.txt (hunk 1 above)
Comment (jmarrec) [reply src/Slab/CMakeLists.txt]: Good catch, a revert conflict likely around https://github.com/NREL/EnergyPlus/commit/26b0c38fb6f09f5540946e5508cb3a1e1ee33bc3 . I made the replacement

**Comment by jmarrec** [comment]
LGTM @rraustad

**Comment by rraustad** [comment]
Thanks @jmarrec I do appreciate it.

**Comment by jmarrec** [comment]
@rraustad I've merged develop in. Assuming CI is happy, then someone should merge it (I don't have rights to do so)

**Comment by mjwitte** [comment]
@rraustad @jmarrec Thanks. Confirmed this fixes the issue:
```
 Working on year            7
STOP GroundTempCalc Completed Successfully.
```
Merging.

**Pull request merged**